packages/webos/VoiceControl/VoiceControl.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class VoiceControl extends React.Component {
+	static propTypes = {
+		voiceHandler: PropTypes.func.isRequired,
+		voiceIntent: PropTypes.string.isRequired,
+		voiceDisabled: PropTypes.bool,
+		voiceLabel: PropTypes.string,
+		voiceTitle: PropTypes.string
+	}
+
+	componentWillUnmount () {
+		this.removeVoiceListener();
+	}
+
+	addVoiceListener = () => {
+		if (this.node) this.node.addEventListener('onvoice', this.handleVoice);
+	}
+
+	removeVoiceListener = () => {
+		if (this.node) this.node.removeEventListener('onvoice', this.handleVoice);
+	}
+
+	handleVoice = (e) => {
+		if (this.props.voiceHandler) {
+			this.props.voiceHandler(e.detail);
+		}
+	}
+
+	getRef = (ref) => {
+		this.node = ref;
+		this.addVoiceListener();
+	}
+
+	render () {
+		const {voiceDisabled, voiceIntent, voiceLabel, voiceTitle, ...props} = this.props;
+		delete props.voiceHandler;
+
+		return (
+			<div
+				ref={this.getRef}
+				data-voice-disabled={voiceDisabled}
+				data-voice-intent={voiceIntent}
+				data-voice-label={voiceLabel}
+				data-voice-title={voiceTitle}
+				{...props}
+			/>
+		);
+	}
+}
+
```

Comment:
Removed all unnecessary props. What about this? and I am not sure this is really needed or not. Too simple...
```
class VoiceControl extends React.Component {
           static propTypes = {
                     onVoice: PropTypes.func.isRequired
           }

           componentDidMount () {
                     findDOMNode(this).addEventListener('onvoice', this.onVoice);
           }

           componentWillUnmount () {
                     findDOMNode(this).removeEventListener('onvoice', this.props.onVoice);
           }

           render = () => (this.props.children)
}
```